app/services/notifications/new_comment/send.rb

```diff
@@ -17,27 +17,40 @@ def call
         return if comment.score.negative?
 
         user_ids = Set.new(comment_user_ids + subscribed_user_ids + top_level_user_ids + author_subscriber_user_ids)
+        user_ids.delete(comment.user_id)
 
         json_data = {
           user: user_data(comment.user),
           comment: comment_data(comment)
         }
 
-        targets = []
-        user_ids.delete(comment.user_id).each do |user_id|
+        user_ids.each do |user_id|
           Notification.create(
             user_id: user_id,
             notifiable_id: comment.id,
             notifiable_type: comment.class.name,
             action: nil,
             json_data: json_data,
           )
-
-          targets << "user-notifications-#{user_id}" if User.find_by(id: user_id)&.mobile_comment_notifications
         end
 
-        # Sends the push notification to Pusher Beams channels. Batch is in place to respect Pusher 100 channel limit.
-        targets.each_slice(100) { |batch| send_push_notifications(batch) }
+        targets = User.where(id: user_ids, mobile_comment_notifications: true).order(:id).ids
+
+        # Pusher Beams uses named Pub/Sub channels instead of raw user_ids
+        target_channels = targets.map { |id| "user-notifications-#{id}" }
+        # Sends the push notification to Pusher Beams channels.
+        # Batch is in place to respect Pusher 100 channel limit.
+        target_channels.each_slice(100) { |batch| send_push_notifications(batch) }
+
+        if FeatureFlag.enabled?(:mobile_notifications)
+          # Send PNs using Rpush
+          PushNotifications::Send.call(
+            user_ids: targets,
+            title: "@#{comment.user.username}",
+            body: "re: #{comment.parent_or_root_article.title.strip}",
+            payload: { url: URL.url("/notifications/comments") },
```

Comment:
non blocking: should we use the helper instead of hardcoding the path here?

`URL.url(Rails.application.routes.url_helpers.notifications_path(:comments))`